realm/realm-library/src/main/java/io/realm/internal/SharedRealm.java

```diff
@@ -208,13 +210,16 @@ private SharedRealm(OsRealmConfig osRealmConfig) {
      * {@code SharedRealm} instance with the same {@link OsRealmConfig} which has been created before. Although they
      * are different {@code shared_ptr}, they point to the same {@code SharedGroup} instance. The {@code context} has
      * to be the same one to ensure core's destructor thread safety.
+     * <p>
+     * WARNING: The {@code SharedRealm} pointer holds by this instance is not managed by the
+     * {@link java.lang.ref.PhantomReference}. {@link #delete()} needs to be called to free the memory.
      */
     private SharedRealm(long nativeSharedRealmPtr, OsRealmConfig osRealmConfig) {
         this.nativePtr = nativeSharedRealmPtr;
         this.osRealmConfig = osRealmConfig;
         this.schemaInfo = new OsSchemaInfo(nativeGetSchemaInfo(nativePtr), this);
         this.context = osRealmConfig.getContext();
-        this.context.addReference(this);
```

Comment:
does it mean we shouldn't `this.context.addReference(this);` when `isPhantomRef` is `true`? 

what's the purpose of keeping the `NativeObject` interface if we don't add the reference(`addReference`)

Reply:
See https://github.com/realm/realm-java/pull/5417/files#diff-6e1f09495610b341d4f834a4e13002ccR199

In normal case it will still be managed by phantom reference.
This constructor is a special case and only handled in this file.

Reply:
`*isPhantomRef` is no longer used? ( you removed the `delete` method) ...?

Reply:
I removed both in the yesterday's change.
So the pointer itself will be managed like a normal `SharedRealm` pointer. So there is no chance for users to meet a invalid pointer anymore. (before yeterday's change, there was still a problem.)

See commit https://github.com/realm/realm-java/pull/5417/commits/ed305d2b8525def04887caff6887b2a0d3a3baa2